contracts/src/SafeInternationalHarbour.sol

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: GNU GPLv3
+pragma solidity ^0.8.29;
+
+
+
+/**
+ * @title SafeInternationalHarbour
+ * @notice Permissionless, append‑only registry that lets **any EOA signer** publish Safe
+ *         transactions ("SafeTx") and their signatures. Clients without an off‑chain indexer can reconstruct the full multisig payload with only:
+ *         1. the Safe address;
+ *         2. the target `chainId`;
+ *         3. the Safe `nonce`; and
+ *         4. the current Safe owners set.
+ *
+ * Each unique `safeTxHash` (EIP‑712 digest of the SafeTx struct) is persisted **once** together with
+ * its parameters. Signatures are appended under the composite key
+ * `(signer, safe, chainId, nonce)`, enabling on‑chain, gas‑efficient look‑ups.
+ *
+ * ### ⚠️ Signature‑malleability disclaimer
+ * This implementation deliberately **does not** enforce the EIP‑2 "low‑`s`" rule. If two distinct
+ * `(r,s,v)` values recover to the same address, *both* will be stored. Client code **must**
+ * de‑duplicate if that is undesirable.
+ *
+ * ### ⚠️ Parameter‑collision disclaimer
+ * Transactions are identified **solely** by their `safeTxHash`. Should two *different* parameter
+ * sets ever collide (astronomically unlikely) the first stored version prevails; later submissions
+ * are silently ignored.
```

Comment:
nit: disclaimer is probably not necessary - if this collision is found, Ethereum is broken.